generated/nidmm/nidmm/_attributes.py

```diff
@@ -100,6 +100,44 @@ def __set__(self, session, value):
         return self._underlying_attribute.__set__(session, value.value)
 
 
+class AttributeEnumWithConverter(AttributeEnum):
+    '''Attribute metaclass for enums that use converters.'''
+
+    def __init__(
+        self,
+        underlying_attribute_meta_class,
```

Comment:
What if instead of passing 

1. `underlying_attribute_meta_class`
2. `enum_meta_class`
3. `attribute_id`

_and_ deriving from `AttributeEnum`...

...you instead took in an instance of the underlying attribute?

I think that would make the code simpler to follow and simpler to unit test.

Reply:
Thanks for the suggestion, I have updated `AttributeEnumWithConverter` to use this design in the latest commit.